src/navigation_table.py

```diff
@@ -31,7 +31,8 @@
 _PATH_REGEX = rf"{_WHITESPACE}([${constants.PATH_CHARS}]+){_WHITESPACE}"
 _PUNCTUATION = string.punctuation.replace("/", "\\/")
 _NAVLINK_TITLE_REGEX = rf"[\w\- {_PUNCTUATION}]+?"
-_NAVLINK_LINK_REGEX = r"[\w\/-]*"
+# Link characters according to https://www.rfc-editor.org/rfc/rfc3986#section-2
+_NAVLINK_LINK_REGEX = r"[\w\/._~:/?#\[\]@!$&'*+,;-]*"
```

Comment:
I think rfc3986 defines `"(" | ")"` to be valid characters in the URI, but URI containing brackets might just be a very edge case so we can ignore that.

Reply:
yeh I removed those because it would interfere with the markdown link which puts brackets around the link